thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/formatter/DetectionConfigFormatter.java

```diff
@@ -0,0 +1,142 @@
+    this.metricDAO = metricDAO;
+  }
+
+  @Override
+  public Map<String, Object> format(DetectionConfigDTO config) {
+    Map<String, Object> output = new HashMap<>();
+    output.put(ATTR_ID, config.getId());
+    output.put(ATTR_IS_ACTIVE, config.isActive());
+    output.put(ATTR_CREATED_BY, config.getCreatedBy());
+    output.put(ATTR_UPDATED_BY, config.getUpdatedBy());
+    output.put(ATTR_NAME, config.getName());
+    output.put(ATTR_DESCRIPTION, config.getDescription());
+    output.put(ATTR_YAML, config.getYaml());
+    output.put(ATTR_LAST_TIMESTAMP, config.getLastTimestamp());
+    List<String> metricUrns = extractMetricUrnsFromProperties(config.getProperties());
+    // the metric urns monitored by this detection config
+    output.put(ATTR_METRIC_URNS, metricUrns);
+    // the default window size of the alert details page
+    output.put(ATTR_ALERT_DETAILS_WINDOW_SIZE, getAlertDetailsDefaultWindowSize(config, metricUrns));
+    return output;
+  }
+
+  private List<String> extractMetricUrnsFromProperties(Map<String, Object> properties) {
+    List<String> metricUrns = new ArrayList<>();
+    if (properties.containsKey(PROP_NESTED_METRIC_URNS_KEY)) {
+      metricUrns.addAll(ConfigUtils.getList(properties.get(PROP_NESTED_METRIC_URNS_KEY)));
+    }
+    List<Map<String, Object>> nestedProperties = ConfigUtils.getList(properties.get(PROP_NESTED_PROPERTIES_KEY));
+    // extract the metric urns recursively from the nested properties
+    for (Map<String, Object> nestedProperty : nestedProperties) {
+      metricUrns.addAll(extractMetricUrnsFromProperties(nestedProperty));
+    }
+    return metricUrns;
+  }
+
+  private long getAlertDetailsDefaultWindowSize(DetectionConfigDTO config, List<String> metricUrns) {
+    try {
+      List<TimeGranularity> granularities = getMonitoringGranularities(config);
+      // if monitoring granularities is not set, use the metric granularity to decide the default window
+      if (granularities.isEmpty()) {
+        granularities = metricUrns.stream().map(this::getGranularityForMetricUrn).collect(Collectors.toList());
+      }
+      List<Long> windowSizes =
+          granularities.stream().map(this::getDefaultWindowSizeForGranularity).collect(Collectors.toList());
+      return Collections.min(windowSizes);
+    } catch (Exception e) {
+      LOG.warn("Exception thrown when getting granularities for detection config {}, use default presenting window size", config.getId(), e);
+      return TimeUnit.DAYS.toMillis(30);
+    }
+  }
+
+  private List<TimeGranularity> getMonitoringGranularities(DetectionConfigDTO config) {
+    List<TimeGranularity> monitoringGranularities = new ArrayList<>();
+    for (Object specs : config.getComponentSpecs().values()) {
+      if (((Map<String, Object>) specs).containsKey(PROP_MONITORING_GRANULARITY)) {
+        monitoringGranularities.add(
+            TimeGranularity.fromString(MapUtils.getString((Map<String, Object>) specs, (PROP_MONITORING_GRANULARITY))));
+      }
```

Comment:
What's the behavior if monitoring granularity is not set on all components?

Maybe if monitoring granularity is not set, then add default granularity here?

Reply:
If monitoring granularity is not set, we will get the default detection granularity from the metric's native granularity.